### PR TITLE
SWC-5663: add missing async job types, and add files to the Download …

### PR DESCRIPTION
**Pull request by jay-hodgson**
…Cart in alpha mode

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java [hunk 1]
```diff
@@ -300,6 +300,10 @@
 	public static final String VIEW_COLUMN_MODEL_REQUEST = "/column/view/scope";
 	public static final String SCHEMA_TYPE_VALIDATION = "/schema/type/validation/";
 	public static final String DOWNLOAD_LIST_V2 = "/download/list/query/";
+	public static final String DOWNLOAD_LIST_ADD_V2 = "/download/list/add/";
+	public static final String DOWNLOAD_LIST_PACKAGE_V2 = "/download/list/package/";
+	public static final String DOWNLOAD_LIST_MANIFEST_V2 = "/download/list/package/";
+	public static final String FILE_HANDLE_RESTORE = "/fileHandle/restore/";
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java:306]: Add missing async job endpoints

File: src/main/java/org/sagebionetworks/web/client/widget/entity/file/AddToDownloadList.java [hunk 2]
```diff
@@ -50,12 +54,15 @@
 	int fileCountToAdd;
 	public static final String DOWNLOAD_ACTION_EVENT_NAME = "Download";
 	public static final String FILES_ADDED_TO_DOWNLOAD_LIST_EVENT_NAME = "FilesAddedToDownloadList";
+	CookieProvider cookies;
+	
 	@Inject
-	public AddToDownloadList(AddToDownloadListView view, AsynchronousProgressWidget progress, InlineAsynchronousProgressViewImpl inlineProgressView, PopupUtilsView popupUtilsView, EventBus eventBus, SynapseAlert synAlert, PackageSizeSummary packageSizeSummary, SynapseJavascriptClient jsClient, SynapseJSNIUtils jsniUtils, AuthenticationController authController) {
+	public AddToDownloadList(AddToDownloadListView view, AsynchronousProgressWidget progress, InlineAsynchronousProgressViewImpl inlineProgressView, PopupUtilsView popupUtilsView, EventBus eventBus, SynapseAlert synAlert, PackageSizeSummary packageSizeSummary, SynapseJavascriptClient jsClient, SynapseJSNIUtils jsniUtils, AuthenticationController authController, CookieProvider cookies) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/file/AddToDownloadList.java:60]: AddToDownloadList now uses v2 API if in alpha mode

File: src/main/java/org/sagebionetworks/web/shared/asynch/AsynchType.java [hunk 3]
```diff
@@ -30,12 +38,11 @@
...
-
+			DownloadStorageReportRequest.class, DownloadStorageReportResponse.class), CreateJsonSchema(SCHEMA_TYPE_CREATE, CreateSchemaRequest.class, CreateSchemaResponse.class), ViewColumnModelRequest(VIEW_COLUMN_MODEL_REQUEST, ViewColumnModelRequest.class, ViewColumnModelResponse.class), GetValidationSchema(SCHEMA_TYPE_VALIDATION, GetValidationSchemaRequest.class, GetValidationSchemaResponse.class), QueryDownloadList(DOWNLOAD_LIST_V2, DownloadListQueryRequest.class, DownloadListQueryResponse.class), AddToDownloadList(DOWNLOAD_LIST_ADD_V2, AddToDownloadListRequest.class, AddToDownloadListResponse.class), DownloadPackageList(DOWNLOAD_LIST_PACKAGE_V2, DownloadListPackageRequest.class, DownloadListPackageResponse.class), DownloadListManifest(DOWNLOAD_LIST_MANIFEST_V2, DownloadListManifestRequest.class, DownloadListManifestResponse.class), FileHandleRestore(FILE_HANDLE_RESTORE, FileHandleRestoreRequest.class, FileHandleRestoreResponse.class);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/shared/asynch/AsynchType.java:41]: This fixes the build (there's a test that verifies that we have client-side representations for all Async job types available in the backend, which we didn't!)

File: src/test/java/org/sagebionetworks/web/unitclient/widget/entity/file/AddToDownloadListTest.java [hunk 4]
```diff
@@ -111,6 +119,97 @@ public void setUp() throws Exception {
 		when(mockAuthController.isLoggedIn()).thenReturn(true);
 	}
 
+	@Test
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/entity/file/AddToDownloadListTest.java:122]: Copy the old tests, but put it in alpha mode to verify the Download List V2 request.

**Review by xschildw — APPROVED** [review]

**Pull request merged**